ModelCatalogueCorePlugin/src/groovy/org/modelcatalogue/core/export/inventory/DataModelToDocxExporter.groovy

```diff
@@ -0,0 +1,417 @@
+            'paragraph.heading2' font: [size: 18, bold: true]
+            'paragraph.heading3' font: [size: 16, bold: true]
+            'paragraph.heading4' font: [size: 16]
+            'paragraph.heading5' font: [size: 15]
+            'paragraph.heading6' font: [size: 14]
+            'cell.headerCell' font: [color: '#29BDCA', size: 12.pt, bold: true], background: '#F2F2F2'
+            'cell' font: [size: 10.pt]
+
+        }
+
+    }
+
+    DataModelToDocxExporter(DataModel rootModel, DataClassService dataClassService, Closure customTemplate, imagePath) {
+        initVars(rootModel, dataClassService)
+        this.customTemplate = customTemplate
+        if(imagePath) this.imagePath = imagePath
+    }
+
+    ArrayList getAllVersionsOfModel(DataModel model, List<DataModel> models) {
+        def oldModel = model.supersedes
+        if (oldModel && !models.contains(oldModel.first())) {
+            models.add(0, oldModel.first())
+            getAllVersionsOfModel(oldModel.first(), models)
+        } else {
+            return models
+        }
+    }
+
+    void export(OutputStream outputStream) {
+        log.info "Exporting Data Model ${rootModel.name} (${rootModel.combinedVersion}) to inventory document."
+
+        def dataClasses = dataClassService.getTopLevelDataClasses(DataModelFilter.includes(rootModel)).items
+        def builder = new ModelCatalogueWordDocumentBuilder(outputStream)
+
+        List<DataModel> allVersionsOfModel = getAllVersionsOfModel(rootModel, [rootModel])
+
+
+
+        builder.create {
+            document(template: customTemplate) {
+
+                def organisation = rootModel.ext.get(Metadata.ORGANISATION)
+                def owner = rootModel.ext.get(Metadata.OWNER)
+                byte[] imageData
+                if(imagePath) imageData = new URL(imagePath).bytes
+
+                if (imagePath){
+                    paragraph(style: 'headerImage', align: 'right'){
+                        image(data: imageData, height: 1.366.inches, width: 2.646.inches, name: 'frontpageImage')
+                    }
+                }
+
+                if (organisation) {
+                    paragraph(style: 'title', align: 'center') {
+                        text organisation
+                    }
+                }
+
+                paragraph (style: 'subtitle', align: 'center'){
+                    text "${rootModel.name}"
+                }
+
+                paragraph(style: 'document', margin: [top: 120]) {
+                    text "Version ${rootModel.versionNumber} ${rootModel.status}"
+                    lineBreak()
+                    text SimpleDateFormat.dateInstance.format(new Date())
+                }
+                pageBreak()
+                heading1 'Document Management'
+                paragraph(style: 'document', margin: [top: 20]) {
+                    text 'Document Owner: ', font:[bold:true]
+                    text owner
```

Comment:
When I try to export the NHIC data model in the dev mode I got `NullPointerException`. The problem is misfortunate variable `owner` which actually is one of the properties of [Groovy Closure Object](http://docs.groovy-lang.org/latest/html/api/groovy/lang/Closure.html#getOwner()). Renaming it to `aOwner` or `theOwner` should fix the problem.



```
 2016-03-10 08:02:58,092 [pool-5-thread-4] ERROR core.AssetService  - Exception storing asset with output stream
Message: null
    Line | Method
->>  134 | doCall              in org.modelcatalogue.core.export.inventory.DataModelToDocxExporter$_export_closure3_closure5_closure10
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|     57 | doCall              in com.craigburke.document.core.builder.DocumentBuilder$_invokeMethod_closure1
|     56 | invokeMethod . . .  in com.craigburke.document.core.builder.DocumentBuilder
|    133 | doCall              in org.modelcatalogue.core.export.inventory.DataModelToDocxExporter$_export_closure3_closure5
|     57 | doCall . . . . . .  in com.craigburke.document.core.builder.DocumentBuilder$_invokeMethod_closure1
|     56 | invokeMethod        in com.craigburke.document.core.builder.DocumentBuilder
|    103 | doCall . . . . . .  in org.modelcatalogue.core.export.inventory.DataModelToDocxExporter$_export_closure3
|     57 | doCall              in com.craigburke.document.core.builder.DocumentBuilder$_invokeMethod_closure1
|     56 | invokeMethod . . .  in com.craigburke.document.core.builder.DocumentBuilder
|    102 | export              in org.modelcatalogue.core.export.inventory.DataModelToDocxExporter
|     43 | doCall . . . . . .  in org.modelcatalogue.core.genomics.GenomicsController$_exportGelSpecification_closure2
|    144 | doCall              in org.modelcatalogue.core.AssetService$_storeAssetWithSteam_closure3
|     48 | store . . . . . . . in org.modelcatalogue.core.LocalFilesStorageService
|    138 | storeAssetWithSteam in org.modelcatalogue.core.AssetService
|    200 | doCall . . . . . .  in org.modelcatalogue.core.AssetService$_storeReportAsAsset_closure4_closure5
|     78 | withDefaultAuthorId in org.modelcatalogue.core.audit.AuditService
|    196 | doCall . . . . . .  in org.modelcatalogue.core.AssetService$_storeReportAsAsset_closure4
|     34 | doCall              in grails.plugin.executor.PersistenceContextRunnableWrapper$_run_closure1
|     35 | wrap . . . . . . .  in grails.plugin.executor.PersistenceContextWrapper
|     34 | run                 in grails.plugin.executor.PersistenceContextRunnableWrapper
|    511 | call . . . . . . .  in java.util.concurrent.Executors$RunnableAdapter
|    266 | run                 in java.util.concurrent.FutureTask
|   1142 | runWorker . . . . . in java.util.concurrent.ThreadPoolExecutor
|    617 | run                 in java.util.concurrent.ThreadPoolExecutor$Worker
^    744 | run . . . . . . . . in java.lang.Thread
```